luigi/scheduler.py

```diff
@@ -830,6 +836,26 @@ def _reset_orphaned_batch_running_tasks(self, worker_id):
         for task in orphaned_tasks:
             self._state.set_status(task, PENDING)
 
+    @rpc_method()
+    def count_pending(self, worker):
+        worker_id, worker = worker, self._state.get_worker(worker)
+
+        num_pending, num_unique_pending, num_last_scheduled = 0, 0, 0
+
+        worker_starts = self._state.get_worker_starts()
+        for task in worker.get_pending_tasks(self._state):
+            num_pending += 1
+            num_unique_pending += int(len(task.workers) == 1)
+            if task.workers:
+                last_scheduled = max(task.workers, key=worker_starts.get)
+                num_last_scheduled += int(last_scheduled == worker_id)
+        return {
+            'n_pending_tasks': num_pending,
+            'n_unique_pending': num_unique_pending,
+            'n_pending_last_scheduled': num_last_scheduled,
+            'worker_state': worker.state,
+        }
```

Comment:
Yeah, I bet we could actually just call count_pending at the end of get_work rather than doing it inline without significantly affecting runtimes. Pulling the logic out and sharing it between both functions would probably be more awkward, confusing and slow. If we add running_tasks to count_pending, we could even exit the get_work loop early when we pick a best_task that isn't batchable.